enctests/testframework/main.py

```diff
@@ -348,6 +348,74 @@ def vmaf_compare(source_clip, test_ref, testname, comparisontestinfo, source_pat
     enc_meta = get_test_metadata_dict(test_ref)
     enc_meta['results'].update(results)
 
+def identity_compare(source_dict, source_clip, test_ref, testname, comparisontestinfo, source_path, distorted, log_file_object):
+    """
+    Compare the sourceclip to the test_ref using ffmpeg identity.
+    We allow it to compare one wedge to another.
+
+    compare_movie -- The image we are comparing to, which by default we assume is the output of the first test.
+    test_ref    -- The generated movie we are testing.
+    testname    -- The testname we are running.
+    comparisontestinfo -- other parameters that can be used to configure the test.
+
+    This creates a results dictionary with the following parameters:
+    mean_error 
+    rms_error
+    peak_snr
+    max_error
+    result - Was the test able to run (Completed = Yes)
+    success - Boolean, was the test a success. 
+    """
+    if source_clip not in source_dict:
+        source_dict[source_clip] = distorted
+        enc_meta = get_test_metadata_dict(test_ref)
+        result = {'success': True, 'result': "Using image as reference"}
+
+        enc_meta['results'].update(result)
+        return
+    compare_movie = source_dict[source_clip]
+
+
+    default_app_template = "{ffmpeg_bin} -nostats -hide_banner -i {originalfile} -i {newfile} -lavfi identity -f null -"
+    apptemplate = comparisontestinfo.get("testtemplate", default_app_template)
+
+    if "compare_image" in comparisontestinfo:
+        compare_movie = comparisontestinfo.get("compare_image")
+
+    if not distorted.exists():
+        result = {'success': False,
+            'testresult': "No movie generated."
+        }
+    else:
+        cmd = apptemplate.format(originalfile=compare_movie, 
+                                 newfile=distorted.as_posix(), ffmpeg_bin=FFMPEG_BIN)
+        print(f"\n\identity command: {cmd}", file=log_file_object)
+        
+        result = subprocess.run(shlex.split(cmd), 
+                                check=False, 
+                                capture_output=True, text=True
+                                )
+        lines = result.stderr.splitlines()
+        print("\n".join(lines), file=log_file_object)
+        if len(lines) < 2:
+            result['result'] = "Unable to run ffmpeg"
+            result['success'] = False
```

Comment:
I'm sure this works, but I assume there's a return code available from the subprocess we could use as well.

Reply:
There should be a new patch to use the return code.

Reply:
I don't think the patch made it through, but I don't think this should hold back the PR